src/machine.rs

```diff
@@ -349,10 +349,23 @@ pub struct Evaluator<'mir, 'tcx> {
 
     /// Equivalent setting as RUST_BACKTRACE on encountering an error.
     pub(crate) backtrace_style: BacktraceStyle,
+
+    /// Crates which are considered local for the purposes of error reporting
+    pub(crate) local_crates: Vec<CrateNum>,
 }
 
 impl<'mir, 'tcx> Evaluator<'mir, 'tcx> {
     pub(crate) fn new(config: &MiriConfig, layout_cx: LayoutCx<'tcx, TyCtxt<'tcx>>) -> Self {
+        // Convert the local crate names from the passed-in config into CrateNums so that they can
+        // be looked up quickly during execution
+        let mut local_crates = Vec::new();
+        for num in layout_cx.tcx.crates(()) {
+            let name = layout_cx.tcx.crate_name(*num).as_str().to_string();
+            if config.local_crates.contains(&name) {
+                local_crates.push(*num);
+            }
+        }
```

Comment:
I don't really know where to put this logic. But it's really nice to have because it decouples this check from the Context, and it has huge performance implications for the other things I want to do.

Reply:
If in doubt, put it into `helpers.rs`...

Reply:
Oh there is some persistent state here, now I see what you mean. Seems fine.